graphstorm-processing/pyproject.toml

```diff
@@ -10,7 +10,7 @@ authors = [
 
 [tool.poetry.dependencies]
 python = "~3.9.12"
-pyspark = ">=3.3.0, < 3.5.0"
+pyspark = ">=3.3.0, <= 3.5.0"
```

Comment:
I see spark has 3.5.1 version available on Feb.23rd, 2024, may we change here to 3.5.1?

Reply:
Good point, I'll change to `<3.6.0`